src/Stratis.Bitcoin.Features.Consensus/CoinViews/CoinviewHelper.cs

```diff
@@ -20,8 +20,10 @@ public OutPoint[] GetIdsToFetch(Block block, bool enforceBIP30)
             {
                 if (enforceBIP30)
                 {
+                    // Calculate the hash outside the loop.
```

Comment:
I vaguely recall we cache the hash once it is computed for the first time for a transaction. But there is no harm structuring the loop in this more sensible way.

Reply:
I did actually have a look but could not find any caching. The reason could be that there is no cut-off really to establish when a transaction had been completely built.